.github/workflows/test.yml

```diff
@@ -107,3 +107,50 @@ jobs:
 
       - name: Upload coverage
         uses: codecov/codecov-action@v2
+
+  test-pandas-versions:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        pandas-version:
+          - ">=1.1,<1.2"
+          - ">=1.2,<1.3"
+          - ">=1.3,<1.4"
+          - ">=1.4"
+      fail-fast: false
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python
+        id: setup-python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+
+      - name: Load cached venv
+        id: cached-poetry-dependencies
+        uses: actions/cache@v2
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-3.8-${{ hashFiles('**/poetry.lock') }}
+
+      - name: Install dependencies
```

Comment:
<img width="374" alt="image" src="https://user-images.githubusercontent.com/33594071/188332279-c0729311-f843-4626-9394-58ca86e020a9.png">

As a result of caching we will run the same test case without pandas installing.